src/Core/src/Platform/iOS/KeyboardAutoManagerScroll.cs

```diff
@@ -276,17 +277,48 @@ internal static async Task AdjustPositionDebounce()
 
 		var entranceCount = DebounceCount;
 
-		await Task.Delay(10);
+		// If we are going to a new view that has an InputAccessoryView
```

Comment:
If I'm following this code correctly it looks like we've eliminated the `Task.Delay(10)` from the original code. Is that intentional?  All of the added delays are inside `if` statements so there's a possibility that we'll just get directly to the `AdjustPosition` without any debouncing happening.

Reply:
Hey @PureWeen, sorry! Yeah I did remove the Task.Delay that happened every time. It was not consistent (some times things would need a longer delay - over 10 ms for editors some times). I tried upping it for everything, but also realized that removing this delay caused the scrolling animation with the keyboard to be much smoother. Also without it, there are usually two calls to the AdjustPosition (one from the entry/editor being focused and one from the keyboard coming up), but the scrolling requested for each work together. I instead changed this "always delay" to only be delays for things that need it like changing InputAccessoryViews.